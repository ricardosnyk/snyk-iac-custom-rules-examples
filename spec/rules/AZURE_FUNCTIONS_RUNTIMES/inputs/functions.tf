provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_windows_function_app" "valid" {
  name                = "example-windows-function-app"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  storage_account_name        = azurerm_storage_account.valid.name
  storage_account_access_key  = azurerm_storage_account.valid.primary_access_key
  service_plan_id             = azurerm_service_plan.valid.id


  site_config {
    application_stack {
      powershell_core_version = "7"
    }
  }
}

resource "azurerm_linux_function_app" "valid1" {
  name                = "example-linux-function-app"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  storage_account_name          = azurerm_storage_account.example.name
  storage_account_access_key    = azurerm_storage_account.example.primary_access_key
  service_plan_id               = azurerm_service_plan.example.id

    site_config {
      application_stack {
        node_version = "14"
    }
  }
}

resource "azurerm_windows_function_app" "invalid1" {
  name                = "example-windows-function-app1"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  storage_account_name        = azurerm_storage_account.example.name
  storage_account_access_key  = azurerm_storage_account.example.primary_access_key
  service_plan_id             = azurerm_service_plan.example.id
  
    site_config {
      application_stack {
        use_custom_runtime = "custom"
    }
  }
}

resource "azurerm_linux_function_app" "invalid" {
  name                = "example-linux-function-app1"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  storage_account_name          = azurerm_storage_account.example.name
  storage_account_access_key    = azurerm_storage_account.example.primary_access_key
  service_plan_id               = azurerm_service_plan.example.id

    site_config {
      application_stack {
        python_version = "3.10"
    }
  }
}