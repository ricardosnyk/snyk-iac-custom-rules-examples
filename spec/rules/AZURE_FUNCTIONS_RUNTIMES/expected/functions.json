[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "azurerm_linux_function_app.invalid",
    "resource_namespace": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
    "resource_type": "azurerm_linux_function_app",
    "severity": "high",
    "resources": [
      {
        "id": "azurerm_linux_function_app.invalid",
        "type": "azurerm_linux_function_app",
        "namespace": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
        "location": [
          {
            "filepath": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
            "line": 59,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "site_config",
              0,
              "application_stack",
              0
            ],
            "location": {
              "filepath": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
              "line": 69,
              "column": 7
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "azurerm_windows_function_app.invalid1",
    "resource_namespace": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
    "resource_type": "azurerm_windows_function_app",
    "severity": "high",
    "resources": [
      {
        "id": "azurerm_windows_function_app.invalid1",
        "type": "azurerm_windows_function_app",
        "namespace": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
        "location": [
          {
            "filepath": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
            "line": 43,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "site_config",
              0,
              "application_stack",
              0
            ],
            "location": {
              "filepath": "spec/rules/AZURE_FUNCTIONS_RUNTIMES/inputs/functions.tf",
              "line": 53,
              "column": 7
            }
          }
        ]
      }
    ]
  }
]